### PR TITLE
feat(core): patch `/user/:userId/custom-data`

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(core): patch `/user/:userId/custom-data`

## Linear Issue Reference

* LOG-805

## Testing

Before (GET):
customDataBefore
Patch with invalid custom data:
custom-old-patch
Patch with correct custom data:
custom-new-patch
Partial update (PATCH):
custom-partial-update
After (GET):
custom-get-after

### CODE REVIEW

File: packages/core/src/routes/user.ts [hunk 1]
```diff
@@ -66,4 +66,31 @@ export default function userRoutes<T extends AnonymousRouter>(router: T) {
       return next();
     }
   );
+
+  /**
+   * User Custom Data
+   */
```
Comment (gao-sun) [inline packages/core/src/routes/user.ts:72]: not very meaningful?

File: packages/core/src/routes/user.ts [hunk 2]
```diff
@@ -66,4 +66,31 @@ export default function userRoutes<T extends AnonymousRouter>(router: T) {
...
+  router.patch(
+    '/user/:userId/custom-data',
```
Comment (gao-sun) [inline packages/core/src/routes/user.ts:74]: should be merged into `adminUser` route? @wangsijie @simeng-li

File: packages/core/src/routes/user.ts (hunk 1 above)
Comment (xiaoyijun) [reply packages/core/src/routes/user.ts]: updated

File: packages/core/src/routes/user.ts (hunk 2 above)
Comment (xiaoyijun) [reply packages/core/src/routes/user.ts:71]: yes, my fault.

File: packages/core/src/routes/admin-user.ts [hunk 3]
```diff
@@ -50,4 +50,28 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
       return next();
     }
   );
+
+  router.patch(
+    '/user/:userId/custom-data',
```
Comment (wangsijie) [inline packages/core/src/routes/admin-user.ts:55]: users?

File: packages/core/src/routes/admin-user.ts [hunk 4]
```diff
@@ -50,4 +50,28 @@ export default function adminUserRoutes<T extends AuthedRouter>(router: T) {
...
+    koaGuard({
+      params: object({ userId: string().min(1) }),
+      body: customDataGuard,
```
Comment (wangsijie) [inline packages/core/src/routes/admin-user.ts:58]: in a PATCH request, we usually use `{ customData: {} }`

File: packages/core/src/routes/admin-user.ts (hunk 3 above)
Comment (xiaoyijun) [reply packages/core/src/routes/admin-user.ts]: updated

File: packages/core/src/routes/admin-user.ts (hunk 4 above)
Comment (xiaoyijun) [reply packages/core/src/routes/admin-user.ts]: both the code & the PR summary is updated

**Review by wangsijie — APPROVED** [review]

**Review by gao-sun — APPROVED** [review]

**Pull request merged**